plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -121,18 +145,21 @@ func TestMetricsServerError(t *testing.T) {
 	defer mockPrometheus.Close()
 
 	logger := zap.NewNop()
+	tracer, exp, closer := tracerProvider(t)
 	address := mockPrometheus.Listener.Addr().String()
-	reader, err := NewMetricsReader(logger, config.Configuration{
+	reader, err := NewMetricsReader(config.Configuration{
 		ServerURL:      "http://" + address,
 		ConnectTimeout: defaultTimeout,
-	})
+	}, logger, tracer)
 	require.NoError(t, err)
 
 	m, err := reader.GetCallRates(context.Background(), &params)
 	assert.NotNil(t, m)
-
+	assert.NotEmpty(t, exp.GetSpans(), "No spans recorded during the test.")
```

Comment:
1. this code would read better if the message was "spans should be recorded". Otherwise when reading this code all I see is "no spans recorded" - is that good/expected?.
2. why not validate the exact number of spans you expect?